File: src/transpiler/ast/Attribute.cpp
#include "Attribute.hpp"

using namespace swift2ts::ast;

Attribute::Attribute(size_t line, size_t col, IdentifierPtr name)
    : Node(line, col), name(name)
{
}

std::any Attribute::accept(const ASTVisitor &visitor) const
{
    return visitor.visit(*this);
}

File: src/transpiler/ast/Attribute.ts
/** @module "transpiler/ast" */

import { mixin } from '../../Decorator';
import { ListLike, Node } from './Node';

/**
 * Attribute node.
 * @todo
 */
export class Attribute extends Node {
    // todo
}

/**
 * Attributes node.
 */
@mixin(ListLike)
export class Attributes extends Node implements ListLike<Attribute> {
    public children: Attribute[];
}
